astropy/io/fits/tests/test_image.py

```diff
@@ -1790,6 +1790,31 @@ def test_compressed_integers(self, dtype):
         new = fits.getdata(testfile)
         np.testing.assert_array_equal(data, new)
 
+    def test_write_non_contiguous_data(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/2150
+        """
+        orig = np.arange(100, dtype=float).reshape((10, 10), order='f')
+        primary = fits.PrimaryHDU()
+        hdu = fits.CompImageHDU(orig)
+        hdulist = fits.HDUList([primary, hdu])
+        hdulist.writeto(self.temp('test.fits'))
+
+        actual = fits.getdata(self.temp('test.fits'))
+        assert_equal(orig, actual)
+
+    def test_slice_and_write_comp_hdu(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/9955
+        """
+        with fits.open(self.data('comp.fits')) as hdul:
+            hdul[1].data = hdul[1].data[:200, :100]
+            hdul[1].writeto(self.temp('test.fits'))
```

Comment:
Could you assert here that it's non-contiguous before?

Reply:
Done.